gcloud-java-compute/src/test/java/com/google/gcloud/compute/it/ITComputeTest.java

```diff
@@ -855,4 +875,464 @@ public void testListGlobalAddresses() throws InterruptedException {
+    compute.delete(firstDiskId);
+    compute.delete(secondDiskId);
+    compute.deleteSnapshot(firstSnapshotId);
+    compute.deleteSnapshot(secondSnapshotId);
+  }
+
+  @Test
+  public void testAggregatedListDisks() throws InterruptedException {
+    String prefix = BASE_RESOURCE_NAME + "list-aggregated-disk";
+    String[] diskZones = {"us-central1-a", "us-east1-c"};
+    String[] diskNames = {prefix + "1", prefix + "2"};
+    DiskId firstDiskId = DiskId.of(diskZones[0], diskNames[0]);
+    DiskId secondDiskId = DiskId.of(diskZones[1], diskNames[1]);
+    DiskConfiguration configuration =
+        StandardDiskConfiguration.of(DiskTypeId.of(ZONE, "pd-ssd"), 100L);
+    Operation firstOperation = compute.create(DiskInfo.of(firstDiskId, configuration));
+    Operation secondOperation = compute.create(DiskInfo.of(secondDiskId, configuration));
+    while (!firstOperation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    while (!secondOperation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    Set<String> zoneSet = ImmutableSet.copyOf(diskZones);
+    Set<String> diskSet = ImmutableSet.copyOf(diskNames);
+    Compute.DiskFilter diskFilter =
+        Compute.DiskFilter.equals(Compute.DiskField.NAME, prefix + "\\d");
+    Page<Disk> diskPage = compute.listDisks(Compute.DiskAggregatedListOption.filter(diskFilter));
+    Iterator<Disk> diskIterator = diskPage.iterateAll();
+    int count = 0;
+    while (diskIterator.hasNext()) {
+      Disk remoteDisk = diskIterator.next();
+      assertTrue(zoneSet.contains(remoteDisk.diskId().zone()));
+      assertTrue(diskSet.contains(remoteDisk.diskId().disk()));
+      assertEquals(DiskInfo.CreationStatus.READY, remoteDisk.creationStatus());
+      assertNotNull(remoteDisk.creationTimestamp());
+      assertNotNull(remoteDisk.id());
+      assertTrue(remoteDisk.configuration() instanceof StandardDiskConfiguration);
+      assertEquals(100L, (long) remoteDisk.configuration().sizeGb());
+      assertEquals("pd-ssd", remoteDisk.configuration().diskType().diskType());
+      assertEquals(DiskConfiguration.Type.STANDARD, remoteDisk.configuration().type());
+      count++;
+    }
+    assertEquals(2, count);
+    compute.delete(firstDiskId);
+    compute.delete(secondDiskId);
+  }
+
+  @Test
+  public void testCreateGetAndDeprecateImage() throws InterruptedException {
+    String diskName = BASE_RESOURCE_NAME + "create-and-get-image-disk";
+    String imageName = BASE_RESOURCE_NAME + "create-and-get-image";
+    DiskId diskId = DiskId.of(ZONE, diskName);
+    ImageId imageId = ImageId.of(imageName);
+    DiskInfo diskInfo =
+        DiskInfo.of(diskId, StandardDiskConfiguration.of(DiskTypeId.of(ZONE, "pd-ssd"), 100L));
+    Operation operation = compute.create(diskInfo);
+    while (!operation.isDone()) {
+      Thread.sleep(1000L);
+    }
+    Disk remoteDisk = compute.get(diskId);
+    ImageInfo imageInfo = ImageInfo.of(imageId, DiskImageConfiguration.of(diskId));
+    operation = compute.create(imageInfo);
```

Comment:
Yes but it's not that easy. You need a raw image to be loaded in cloud storage. Images are quite big and we can not load them on the go in our ITs. Having the raw image loaded in one of our buckets makes the ITs non-repeatable by users. Given this I decided to skip that test.